ibazel/ibazel.go

```diff
@@ -140,19 +144,31 @@ func (i *IBazel) SetBazelArgs(args []string) {
 	i.bazelArgs = args
 }
 
-func (i *IBazel) SetLiveReload(liveReload bool) {
-	i.liveReload = liveReload
-}
-
 func (i *IBazel) SetDebounceDuration(debounceDuration time.Duration) {
 	i.debounceDuration = debounceDuration
 }
 
 func (i *IBazel) Cleanup() {
 	i.buildFileWatcher.Close()
 	i.sourceFileWatcher.Close()
-	if i.lrserver != nil {
-		i.lrserver.Close()
+	for _, l := range i.lifecycleListeners {
+		l.Cleanup()
+	}
+}
+
+func (i *IBazel) targetDecider(rule *blaze_query.Rule) {
+	for _, l := range i.lifecycleListeners {
+		l.TargetDecider(rule)
```

Comment:
Should target decider return a bool of whether it wants to receive events? Otherwise each lifecycle listener will need to maintain state and have logic to perform a noop on BeforeEvent and AfterEvent calls

Reply:
Yes, that seems more correct than this. I think that might get a little complicated tracking that in a generic form so I'm going to return to it a little later. Todo added with context of why I don't wanna do it right now.